board/gpio.md

```diff
@@ -3,70 +3,97 @@ GPIO API
 
 The GPIO (General Purpose Input & Output) API supports digital pins.
 
-The API object
---------------
-GPIO pin functionality is exposed by the [`GPIO`](#gpio) object that can be obtained by using the [gpio() method of the `Board` API](./README.md/#gpio). See also the [Web IDL](./webidl.md).
-
 On certain boards GPIO pins may be grouped into ports (e.g. 8, 16 or 32 pins), read and written as registers by the controller.
 
 On certain boards, analog pins can also be used as GPIO.
 
-<a name="gpio"></a>
-### The `GPIO` interface
-Represents the properties and methods that expose GPIO functionality. The `GPIO` object implements the [`EventEmitter`](../README.md/#events) interface, and extends the [`Pin`](./README.md/#pin) object.
+<a name="apiobject"></a>
+### The GPIO API object
+When requiring `"gpio"`, the following steps are run:
+- If there is no permission for using the functionality, throw `SecurityError`.
+- If the AIO functionality is not supported on the board, throw `"NotSupportedError"`.
+- Return an object that implements the following methods.
+
+| Method              | Description      |
+| ---                 | ---              |
+| [`open()`](#open)   | open GPIO pin    |
+| [`port()`](#port)   | open GPIO port   |
+
+See also the [Web IDL](./webidl.md) definition.
+
+The following dictionary is used for initializing GPIO pins and ports.
+<a name="gpiooptions"></a>
 
 | Property   | Type   | Optional | Default value | Represents |
 | ---        | ---    | ---      | ---           | ---        |
-| `pin`      | String or Number | no | `undefined`   | board name for the pin |
+| `pin `     | String or Number | no | `undefined`   | pin name |
+| `mapping`  | String | no | `"system"`   | pin mapping |
```

Comment:
Such a requirement is very hard to make normative, and it is usually discouraged. If the driving concern is to get rid of the `mapping` property, then we can get rid of it until it proves to be necessary, but then the pin names become completely opaque to the app code,  platform dependent, but without a specified policy, that is, "if you assume these strings have a meaning, you are on your own to find their docs".

Let's suppose we get rid of `mapping`. We could still require implementing board namespaces, i.e. support for the labels that are printed on the board. So the developer would have the guarantee that it's enough to look at the board, and those names would work in this API.
I agree that would also be equally hard to make normative, although it would be a nice-to-have feature.

So let's go by middle ground and remove `mapping` with the note above about de-scoping pin name semantics from this API.